src/main/java/org/broadinstitute/hellbender/tools/exome/IntervalBackedExonDB.java

```diff
@@ -0,0 +1,363 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import org.broadinstitute.hellbender.utils.GenomeLoc;
+import org.broadinstitute.hellbender.utils.IndexRange;
+
+import java.util.*;
+
+/**
+ * Exon data-base based on a list of intervals.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+public class IntervalBackedExonDB implements ExonDB<GenomeLoc> {
+
+    /**
+     * Map from interval name to interval object.
+     */
+    private final Map<String,GenomeLoc> intervalsByName;
+
+    /**
+     * Sorted list of intervals.
+     */
+    private final List<GenomeLoc> sortedIntervals;
```

Comment:
I wonder if `SortedSet` would not be a better type here to make clear that this is collection is sorted. Do you need random access to the data?